feature-libs/storefinder/core/service/google-map-renderer.service.ts

```diff
@@ -35,18 +36,31 @@ export class GoogleMapRendererService {
     locations: any[],
     selectMarkerHandler?: Function
   ): void {
-    if (Object.entries(locations[Object.keys(locations)[0]]).length > 0) {
-      if (this.googleMap === null) {
-        this.scriptLoader.embedScript({
-          src: this.config.googleMaps.apiUrl,
-          params: { key: this.config.googleMaps.apiKey },
-          attributes: { type: 'text/javascript' },
-          callback: () => {
-            this.drawMap(mapElement, locations, selectMarkerHandler);
-          },
-        });
-      } else {
-        this.drawMap(mapElement, locations, selectMarkerHandler);
+    if (this.config.googleMaps?.apiKey) {
+      if (Object.entries(locations[Object.keys(locations)[0]]).length > 0) {
+        if (this.googleMap === null) {
+          const apiKey =
+            this.config.googleMaps.apiKey === GOOGLE_MAPS_DEVELOPMENT_KEY_CONFIG
```

Comment:
from discussion with Patrick: we keep the dev key available also when Spartacus run in prod mode. It allows our demo instances to display the map.
Compliancy requirement to 'NOT call any third-party server when api key is NOT provided' is fulfilled.